.github/workflows/delete-self-hosted-runner.yml

```diff
@@ -0,0 +1,35 @@
+name: delete azure-self-hosted-runner
+run-name: Delete ${{ inputs.runner_name }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      runner_name:
+        type: string
+        required: true
+        description: The name of the runner that needs to be deleted
+
+env:
+  ACTIONS_RUNNER_NAME: ${{ github.event.inputs.runner_name }}
+
+# The following secrets are required for this workflow to run:
+# AZURE_CREDENTIALS - Credentials for the Azure CLI. It's recommended to set up a resource
+#                     group specifically for self-hosted Actions Runners.
+#   az ad sp create-for-rbac --name "{YOUR_DESCRIPTIVE_NAME_HERE}" --role contributor \
+#     --scopes /subscriptions/{SUBSCRIPTION_ID_HERE}/resourceGroups/{RESOURCE_GROUP_HERE} \
+#     --sdk-auth
+# AZURE_RESOURCE_GROUP - Resource group to create the runner(s) in
+jobs:
+  delete-runner:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Azure Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    - name: Delete VM '${{ env.ACTIONS_RUNNER_NAME }}'
+      uses: azure/CLI@v1
+      with:
+        azcliversion: 2.43.0
+        inlineScript: |
+          az vm delete -n "$ACTIONS_RUNNER_NAME" -g ${{ secrets.AZURE_RESOURCE_GROUP }} --yes
```

Comment:
This works, kind of. But it leaves two resources behind, the network security group and the virtual network:

![image](https://user-images.githubusercontent.com/127790/211604814-aa64c8bd-3107-403d-9b98-05b1f61e6d66.png)

@dennisameling you mentioned previously that we could avoid that by reusing a single resource? Or is there a way to pass one or two `deleteOption`s so that these resources are deleted, too, when the VM is deleted?

Reply:
Let's see whether 4c2ba65 works.

Reply:
Looks like this is a known issue: https://github.com/Azure/azure-cli/issues/22158

We might need 1-2 additional `az` CLI calls to find the NSG/Vnet that are attached to the VM and delete them in follow-up calls.

Reply:
It also looks like `deleteOption` isn't a supported property on `Microsoft.Network/networkSecurityGroups` and `Microsoft.Network/virtualNetworks`:

https://learn.microsoft.com/en-us/azure/templates/microsoft.network/networksecuritygroups?pivots=deployment-language-arm-template#networksecuritygrouppropertiesformat

https://learn.microsoft.com/en-us/azure/templates/microsoft.network/virtualnetworks?pivots=deployment-language-arm-template

Reply:
Indeed, [it does not work](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/3885243390/jobs/6628872716#step:7:21).

> We might need 1-2 additional `az` CLI calls to find the NSG/Vnet that are attached to the VM and delete them in follow-up calls.

I guess we need 2 additional CLI calls, but we know the exact name i.e. we do not have to identify the resources via a separate call.